addons/analytic/models/analytic_plan.py

```diff
@@ -135,19 +135,21 @@ def get_relevant_plans(self, **kwargs):
         list_plans = []
         set_plan_ids = {}
         company_id = kwargs.get('company_id', self.env.company.id)
-        all_plans = self.search([('parent_id', '=', False), '|', ('account_ids', '!=', False), ('children_ids.account_ids', '!=', False),
-                                 '|', ('company_id', '=', company_id), ('company_id', '=', False)])
+        all_plans = self.search([('account_ids', '!=', False), '|', ('company_id', '=', company_id), ('company_id', '=', False)])
         for plan in all_plans:
-            applicability = plan._get_applicability(**kwargs)
+            root_plan = plan if not plan.parent_id else self.env['account.analytic.plan'].browse(int(plan.parent_path.split('/')[0]))
+            if root_plan.id in [plan.get('id') for plan in list_plans]:
```

Comment:
I don't know if we prefer to create a set beforehand, where we just add ids of plans that are in `list_plans`, to avoid doing this list comprehension for each plan, as we should have a small-ish set of plans but I will let @william-andre decide

Reply:
An easy solution is to use a `dict[root_plan_id, values]`.

Note that for unavailable plans, you will call `_get_applicability` every single time too.

Another solution would be something like this
```python
record_account_ids = kwargs.get('existing_account_ids', [])
forced_plans = self.env['account.analytic.account'].browse(record_account_ids).mapped('root_plan_id')
all_plans = self.search([
    ('account_ids', '!=', False), 
    '|', ('company_id', '=', company_id), ('company_id', '=', False),
])
root_plans = self.browse({
    plan.root_plan_id.id
    for plan in all_plans
    if plan._get_applicability(**kwargs) != 'unavailable' or plan in forced_plans
})
return sorted([
    {...}
    for root_plan in root_plans
], key=lambda d: (d['applicability'], d['id']))
```

Reply:
You don't have `root_plan_id` field for a plan. And we still need to loop to get the applicability for root_plans (optional or mandatory) independently of the forced plans

Reply:
Maybe I'm missing something, but if you define "mandatory" in your root plan, you loose this information by hard coding "optionnal" as you suggest

```py
record_account_ids = kwargs.get('existing_account_ids', [])
forced_plans = self.env['account.analytic.account'].browse(record_account_ids).mapped('root_plan_id')
all_plans = self.search([
    ('account_ids', '!=', False), 
    '|', ('company_id', '=', company_id), ('company_id', '=', False),
])
root_plans = self.browse({
    int(plan.parent_path.split('/')[0])
    for plan in all_plans
    if plan._get_applicability(**kwargs) != 'unavailable' or plan in forced_plans
})
return sorted([
    {
        "id": root_plan.id,
        "name": root_plan.name,
        "color": root_plan.color,
        "applicability": 'optional',
        "all_account_count": root_plan.all_account_count
    }
    for root_plan in root_plans
], key=lambda d: (d['applicability'], d['id']))
```